lisp/forge-commands.el

```diff
@@ -50,6 +50,14 @@ Takes the pull-request as only argument and must return a directory."
   :group 'forge
   :type 'function)
 
+(defcustom forge-checkout-reset-existing-branch nil
```

Comment:
I _think_ with the current state of this branch that the default here should be `t`, as that will preserve the existing behavior. `nil` is “dangerous” in that it gives the user an existing branch without letting them know (and without them choosing that behavior). With the existing code they would only be handed a branch without question if it were newly-created.

(With my suggestions below, `nil` _would_ be the correct default.)